apps/client/src/app/components/rule/rule.component.ts

```diff
@@ -16,8 +21,39 @@ import {
 export class RuleComponent implements OnInit {
   @Input() isLoading: boolean;
   @Input() rule: PortfolioReportRule;
+  private unsubscribeSubject = new Subject<void>();
 
-  public constructor() {}
+  public constructor(
+    private dataService: DataService,
+    private changeDetectorRef: ChangeDetectorRef
+  ) {}
 
   public ngOnInit() {}
+
+  public onUpdateAccount(rule: PortfolioReportRule) {
+    let settings: UpdateUserSettingDto = {
+      xRayRules: {
+        [rule.key]: { isActive: !('evaluation' in rule) }
+      }
+    };
+    this.dataService
+      .putUserSetting(settings)
+      .pipe(takeUntil(this.unsubscribeSubject))
+      .subscribe(() => {
+        this.dataService
+          .fetchPortfolioReport()
+          .pipe(takeUntil(this.unsubscribeSubject))
+          .subscribe((report) => {
+            for (const ruleGroup in report.rules) {
```

Comment:
I wonder if it would make more sense to have the data update in the `rules` files instead of here, so here just emit an event an `rules` fetches back the portfolio report, so it updates all rules instead of going through this for loop.

Reply:
Also, querying again the `report` endpoint might not be a good idea, as for some reason that I need to uncover, the response time varies between 100ms and 1 minute. Storing the `isActive` parameter in the server reponse's JSON could be a solution, so we don't need to query the backend when toggling the rule, but it means every rule is evaluated, active or not, when the report is requested.

Reply:
> I wonder if it would make more sense to have the data update in the `rules` files instead of here, so here just emit an event.

Yes, that would be much better. `RuleComponent` (as well as `RulesComponent`) should be "dumb" and just use `@Input()` / `@Output`.

→ `FirePageComponent` should refetch the portfolio report (while showing the loading state)